roles/prune_catalog/tasks/main.yml

```diff
@@ -0,0 +1,126 @@
+---
+- name: "Validate parameters"
+  assert:
+    that:
+      - pc_operators_list is defined
+      - pc_operators_list | type_debug == "list"
+      - pc_source_catalog is defined
+      - pc_destination_catalog is defined
+      - pc_source_catalog | length > 0
+      - pc_destination_catalog | length > 0
+    fail_msg: "At least one of the parameters requirements is missing"
+
+- name: "Check if {{ pc_source_catalog }} is pullable"
+  shell: >
+    if [ -n "$DOCKER_CONFIG" ]; then
+      skopeo inspect --no-tags \
+      --authfile $DOCKER_CONFIG/config.json \
+      --tls-verify=false docker://{{ pc_source_catalog }};
+    else
+      skopeo inspect --no-tags \
+      --tls-verify=false docker://{{ pc_source_catalog }};
+    fi
+  register: skopeo_result
+  retries: 2
+  delay: 10
+  ignore_errors: true
```

Comment:
I think we can stop ignoring errors, and fail if the catalog is not pullable, in the end all the rest of the tasks won't run if this does not succeed.

Reply:
In this case it is important to simple do nothing and not fail, that will allow the agent to detect that that image was not created on that iteration and fallback to the previous minor version. I do not think doing nothing on this role if the image is not pullable.

Reply:
I see, this is more a requirement on the agent, rather than the role itself :thinking: 

Would it be worth doing a conditional for the failure instead?

Reply:
@tonyskapunk removed the ignore_errors. The 2 last jobs test the regular use case and the fallback in case there are operators missing.